src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IConfigurationClientManager.cs

```diff
@@ -5,12 +5,14 @@
 using Azure.Data.AppConfiguration;
 using System;
 using System.Collections.Generic;
-
+using System.Threading;
+using System.Threading.Tasks;
+
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal interface IConfigurationClientManager
     {
-        IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time);
+        Task<IEnumerable<ConfigurationClient>> GetAvailableClients(DateTimeOffset time, CancellationToken cancellationToken);
```

Comment:
IAsyncEnumerable

Reply:
I considered using IAsyncEnumerable, but underlying DnsClient returns `IEnumerable<SrvRecord>`, and after doing shuffling, we synchronize the whole collection. Hence, if just return  IAsyncEnumerable in method, not update the entire upstream calling chain to IAsnyncEnumerable, seems will not get much benifit.